### ISSUE TITLE
[a11y]: [UI Shell] [Header with Actions & Side Nav] -Screen reader should announce the selection status, which is visually indicated by color

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16887]
### Package

@carbon/react

### Browser

Firefox

### Operating System

Windows

### Package version

v1.60.3

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

The category title button is visually indicated as selected, but the screen reader does not announce this selection status.
<img width="295" alt="image" src="https://github.com/carbon-design-system/carbon/assets/63502271/05e40b35-5be1-4eb0-a729-d08615c2d393">
Also observe : https://github.com/carbon-design-system/carbon/issues/16886

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-navigation-actions-and-side-nav

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
